addons/event/models/event_event.py

```diff
@@ -282,47 +298,70 @@ def _compute_start_sale_date(self):
             start_dates = [ticket.start_sale_date for ticket in event.event_ticket_ids if ticket.start_sale_date]
             event.start_sale_date = min(start_dates) if start_dates else False
 
-    @api.onchange('is_online')
-    def _onchange_is_online(self):
-        if self.is_online:
-            self.address_id = False
+    @api.depends('is_online')
+    def _compute_address_id(self):
+        for event in self:
+            if event.is_online:
+                event.address_id = False
+            elif not (event.address_id or event._origin.address_id):
+                event.address_id = self.env.company.partner_id.id
 
-    @api.onchange('event_type_id')
-    def _onchange_type(self):
-        if self.event_type_id:
-            self.seats_max = self.event_type_id.default_registration_max
-            if self.event_type_id.default_registration_max:
-                self.seats_availability = 'limited'
+    @api.depends('event_type_id')
+    def _compute_date_tz(self):
+        for event in self:
+            if event.event_type_id.use_timezone:
+                event.date_tz = event.event_type_id.default_timezone
+            if not event.date_tz and not event._origin.date_tz:
+                event.date_tz = self.env.user.tz or 'UTC'
+
+    @api.depends('event_type_id')
+    def _compute_seats_availability(self):
+        """ Make it separate from ``_compute_from_event_type`` because otherwise
+        a value given at create (see create override) would protect all other fields
+        depending on event type id from being computed (see ``_field_computed``) """
+        for event in self:
+            if event.event_type_id.default_registration_max:
+                event.seats_availability = 'limited'
+            if not event.seats_availability and not event._origin.seats_availability:
+                event.seats_availability = 'unlimited'
+
+    @api.depends('event_type_id')
```

Comment:
Is it ok to only depend on event_type_id and not any of the used sub-fields?

I guess it is since we don't want to recompute those fields when we change something on the event.type itself.
But it's quite tricky...

Reply:
You don't want to reflect changes on event type, indeed. Only when changing the value of event_type_id itself should we compute values based on it.

Reply:
I added a docstring with that explanation.